pkg/syncutil/cmt/cmt.go

```diff
@@ -0,0 +1,106 @@
+package cmt
+
+import (
+	"context"
+	"sync"
+
+	"github.com/go-logr/logr"
+	"github.com/open-policy-agent/frameworks/constraint/pkg/types"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/controller/config/process"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/logging"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/metrics"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/readiness"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/syncutil"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var log = logf.Log.WithName("data-replication").WithValues("metaKind", "CacheManagerTracker")
+
+type CacheManagerTracker struct {
+	lock sync.RWMutex
+
+	opa              syncutil.OpaDataClient
+	syncMetricsCache *syncutil.MetricsCache
+	tracker          *readiness.Tracker
+	processExcluder  *process.Excluder
+
+	// todo acpana -- integrate gvkaggregator
+}
+
+func NewCacheManager(opa syncutil.OpaDataClient, syncMetricsCache *syncutil.MetricsCache, tracker *readiness.Tracker, processExcluder *process.Excluder) *CacheManagerTracker {
+	return &CacheManagerTracker{
+		opa:              opa,
+		syncMetricsCache: syncMetricsCache,
+		tracker:          tracker,
+		processExcluder:  processExcluder,
+	}
+}
+
+func (c *CacheManagerTracker) AddGVKToSync(ctx context.Context, instance *unstructured.Unstructured) (*types.Responses, error) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	isNamespaceExcluded, err := c.processExcluder.IsNamespaceExcluded(process.Sync, instance)
+	if err != nil {
+		log.Error(err, "error while excluding namespaces")
```

Comment:
Can keep `"error while excluding namespaces"` in sync_controller, but this should return a specific error since it's a new module.

Reply:
I don't mind making this change but I want to call out that before/ today, this error is "ignored" or just logged and is not actionable or retried in the sync_controller. See:

https://github.com/open-policy-agent/gatekeeper/blob/9da7226f468344111fa87d10f467d92fdd26c4b2/pkg/controller/sync/sync_controller.go#L190-L194

So technically, this is not the same behavior but I think this is fine for now. We may want to continue execution here in the future even if process exclusion errors out.